test/intfuncs.jl

```diff
@@ -191,6 +191,13 @@ end
 
         @test lcm(T[2, 4, 6]) ⟷ T(12)
     end
+
+    # Issue #55379
+    @test lcm([1//2; 1//2]) === lcm([1//2, 1//2]) === lcm(1//2, 1//2) === 1//2
+    @test gcd(Int[]) === 0
+    @test lcm(Int[]) === 1
+    @test gcd(Rational{Int}[]) === 0//1
```

Comment:
These three tests are unnecessarily strict.

```suggestion
    @test iszero(gcd(Int[]))
    @test iszero(gcd(Rational{Int}[]))
```

Relaxing them should help avoid spurious test failures in the future.

Reply:
If someone changes the return type of `gcd(Int[])` or `gcd(Rational{Int}[])`, I want a test to fail. Similarly, if someone changes the return type or value of `lcm(Int[])`, I want a test to fail. Those would be breaking changes that should not be made accidentally.

I'm adding them because I noticed that #56113 would break all three of these new tests but doesn't break any existing tests; I want to make sure that behavior change is only made if folks are aware that we are changing existing behavior.

Reply:
Changing the return type isn't breaking, as long as the value is correct and the type subtypes the correct abstract type.

Reply:
Making a mathematical operation on `Int`s that used to return `Int`s instead return a different type could introduce errors or correctness bugs based on overflow behavior changes or failed conversions and could also introduce significant performance issues through the introduction of type instability in user code.

Changing `1+1` to return an Int8 is breaking even through Int8 has all the same abstract supertypes as Int. A similar change to `lcm` or `gcd` would be much less impactful because `+` is so much more commonly used but would still be breaking.